test/test_deflate.cc

```
@@ -0,0 +1,141 @@
+    d_stream.next_out = decompbuf;
+    d_stream.avail_out = (uint32_t)size;
+
+    err = PREFIX(inflate)(&d_stream, Z_FINISH);
+    EXPECT_EQ(err, Z_STREAM_END) <<
+        "size: " << size << "\n" <<
+        "level: " << level << "\n" <<
+        "window_bits: " << window_bits << "\n" <<
+        "strategy: " << strategy;
+
+    EXPECT_EQ(d_stream.total_out, size);
+    EXPECT_EQ(memcmp(inbuf, decompbuf, size), 0) << "round-trip data mismatch";
```

Cppcheck (2.20.0)

[warning] If memory allocation fails, then there is a possible null pointer dereference

(nullPointerOutOfMemory)